### PR TITLE
New dashboard final touches

### LINKED ISSUE #3361: Dashboard finishing touches
- When going to https://worksheets.codalab.org/, it should redirect to your user page if you're logged in
- Replace "DASHBOARD" button with a "MY PROFILE" button
- On my (the current user's) profile page, link to the dashboard worksheet from the sidebar

### LINKED ISSUE #3377: No error message when adding worksheet fails
Go to worksheets.codalab.org/users.  Click on 'Add' and type in a worksheet name that already exists.  The error message doesn't show up like it does when you click the + in the top bar.

### LINKED ISSUE #3378: new dashboard: more space, add names
Screenshot from 2021-04-01 22-38-27

Should add more space between [Edit Avatar] and the profile pic.  Or perhaps it's cleaner not to have the button and allow editing when you click on the pic (with a tool tip that says 'Click to edit').

Also, we should show the user's full name.

### PR DESCRIPTION
**Pull request by yuqijin**
### Reasons for making this change
- When the user visits the CodaLab Home Page, he/she will be redirected to his/her profile page.
- On the Navigation bar, `MY PROFILE` button replaced `DASHBOARD` button, which linked to the user's profile page.
- On the profile page, a button `DASHBOARD` linked to the old dashboard worksheet.
- Use the user's avatar to replace the AccountIcon on the Navigation bar.
- The dropdown menu has another menuItem `MY DASHBOARD`, which linked to the old dashboard worksheet.

### Related issues

fixes #3361 
fixes #3378 
fixes #3377 

### Screenshots

**View Dashboard** 
Current view of the new dashboard (fixed comments)
Screen Shot 2021-04-07 at 4 12 11 PM

**Another option for Dashboard button**
Screen Shot 2021-04-07 at 4 02 56 PM

*******
<br />
<br />
<br />

**Demo**
ezgif com-gif-maker

**Viewing other user's profile page**
Screen Shot 2021-04-01 at 10 08 07 PM

**Display both full name and username**
Screen Shot 2021-04-02 at 4 14 53 AM

**Remove `Edit Avatar` button**
ezgif com-gif-maker (1)

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by epicfaace** [comment]
> * On the profile page, a button `DASHBOARD` linked to the old dashboard worksheet.

Can you change this button to say `VIEW MY DASHBOARD` so it's clearer that it's a button? And move it to the sidebar.

Also add more space underneath.

**Comment by yipenghe** [comment]
looks great!!

File: frontend/src/components/NavBar.js [hunk 1]
```diff
@@ -141,6 +142,41 @@ class NavBar extends React.Component<{
         });
     }
 
+    // Fetch the image file represented by the bundle
+    fetchImg(bundleUuid) {
+        if (bundleUuid == null) return;
+        // Set defaults
+        let url = '/rest/bundles/' + bundleUuid + '/contents/blob/';
```
Comment (epicfaace) [inline frontend/src/components/NavBar.js:150]: Interesting method to render the image... could we simplify this by just adding `` directly in the render function instead?

File: tests/ui/README.md [hunk 2]
```diff
@@ -15,8 +15,8 @@ repository, by running `pip install -r requirements.txt`.
 tests and their respective browsers.
     1. Run `sudo ./scripts/test-setup.sh` and place the installed GeckoDriver on your path by running 
     `export PATH=$PATH:/usr/bin/geckodriver`. This is not needed for ChromeDriver.
-    2. An alternative is to simply use [Homebrew](https://brew.sh/) and run `brew install geckodriver && brew cask 
-    install chromedriver`. 
+    2. An alternative is to simply use [Homebrew](https://brew.sh/) and run `brew install geckodriver && brew install
```
Comment (epicfaace) [inline tests/ui/README.md:18]: Why did this change? Was the old command incorrect for homebrew?

**Review by epicfaace — COMMENTED** [review]
Huh, the change in the baseline images are quite drastic: 

image

Any idea why the old baseline images were stuck at v0.5.8 and also at a much larger screen size? I wonder if this indicates some problem with our UI tests... (cc @teetone )

File: tests/ui/README.md (hunk 2 above)
Comment (yuqijin) [reply tests/ui/README.md:18]: The homebrew has updated so the command has been updated. The old command cannot work anymore.

File: frontend/src/components/NavBar.js (hunk 1 above)
Comment (yuqijin) [reply frontend/src/components/NavBar.js:150]: it does not work if we directly use this method in the render function

**Comment by yuqijin** [comment]
> Huh, the change in the baseline images are quite drastic:
> 
> image
> 
> Any idea why the old baseline images were stuck at v0.5.8 and also at a much larger screen size? I wonder if this indicates some problem with our UI tests... (cc @teetone )

The images were stuck at v0.5.8 since the UI tests do not fail in the later versions (difference < 10%), and thus the screenshots have not been updated for a long time. In this version, the DASHBOARD button is updated, which makes the difference larger than 10%, so the screenshots need to be updated.

**Comment by epicfaace** [comment]
Ah, I see, thanks for the explanation!

**Review by epicfaace — APPROVED** [review]

**Comment by epicfaace** [comment]
> The images were stuck at v0.5.8 since the UI tests do not fail in the later versions (difference < 10%), and thus the screenshots have not been updated for a long time. In this version, the DASHBOARD button is updated, which makes the difference larger than 10%, so the screenshots need to be updated.

Just curious, do you know why the window size for the screenshots also changed?

**Comment by yuqijin** [comment]
> > The images were stuck at v0.5.8 since the UI tests do not fail in the later versions (difference < 10%), and thus the screenshots have not been updated for a long time. In this version, the DASHBOARD button is updated, which makes the difference larger than 10%, so the screenshots need to be updated.
> 
> Just curious, do you know why the window size for the screenshots also changed?

It should depends on the UI test file. I inspected the python file and found it depends on the `scrollHeight`. Since it has changed since v0.5.8, that might be the reason.

**Pull request merged**